feat(ci): enable chocolatey workflow for windows binaries

Dr-Electron opened the pull request:
# Description of change

Enabled the workflow to publish the cli to choco again.

## Links to any relevant issues

Fixes #5576.

### Release Notes

- [ ] Protocol:
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [x] CLI: Add CLI to Chocolatey
- [ ] Rust SDK:
- [ ] REST API:

Dr-Electron committed:
feat(actions): re-enable chocolatey

Dr-Electron committed:
Merge branch 'develop' into devx/chocolatey

Dr-Electron committed:
Update requests from choco review

Dr-Electron commented on chocolatey/iota.nuspec:
```diff
@@ -5,22 +5,25 @@ enclosed in quotation marks, you should use an editor that supports UTF-8, not t
+		<authors>iotaledger</authors>
 		<projectUrl>https://iota.org/</projectUrl>
 		<licenseUrl>https://github.com/iotaledger/iota/blob/develop/LICENSE</licenseUrl>
 		<iconUrl>https://assets-global.website-files.com/6425f546844727ce5fb9e5ab/643775f4a15c9a9e10426daa_Iota_Favicon_256.png</iconUrl>
```
```suggestion
		<iconUrl>https://files.iota.org/media/chocolatey/icon.png</iconUrl>
```

Dr-Electron committed:
Update chocolatey/iota.nuspec

Dr-Electron committed:
Merge branch 'develop' into devx/chocolatey

Dr-Electron committed:
Fix nuspec

Dr-Electron committed:
fix version

Dr-Electron committed:
Fix variable

Thoralf-M commented on chocolatey/iota.nuspec:
```diff
@@ -5,23 +5,26 @@ enclosed in quotation marks, you should use an editor that supports UTF-8, not t
 	<files>
-		<file src="../target/release/iota.exe" target="bin/iota.exe" />
-		<file src="../LICENSE" target="bin/LICENSE.txt" />
+		<file src="../tmp/release/iota.exe" target="bin/iota.exe" />
```
This path is really correct?

muXxer replied:
It's coming from `TMP_BUILD_DIR: "./tmp/release"`, but if we ever change the env var, this workflow breaks. Maybe we should add a comment to the env var then?

Dr-Electron replied:
Yeah good point, didn't think about that 🤔
Will add a comment 👍

Dr-Electron replied:
Added a comment in 4ff8ab2

muXxer commented on .github/workflows/release.yml:
```diff
@@ -191,26 +191,28 @@ jobs:
-      #          choco apikey --api-key ${{ secrets.CHOCO_API_KEY }} --source https://push.chocolatey.org/
-      #          choco push iota.${{ env.iota_ref }}.nupkg --source https://push.chocolatey.org/
+      - name: Publish Windows iota binary to Chocolatey
+        if: ${{ matrix.os == 'windows-latest' && github.event_name == 'release' && contains(github.ref, '-rc') }}
```
Why `-rc`?

Dr-Electron replied:
Because that is what we also do with hombrew afaik and if nothing changed @thibault-martinez

thibault-martinez replied:
Correct, we release RCs on homebrew. It's essentially the same binary as stable and is made available earlier to people.

Dr-Electron replied:
Changed. We will release a pre-release (rc) and stable version of both choco and homebrew following our discussion on slack

muXxer commented on .github/workflows/release.yml:
```diff
@@ -191,26 +191,28 @@ jobs:
+          EOF
+
+          # Strip pre-release as we want to publish rc as release version on choco
+          export choco_version=$(echo "${{ needs.generate-tag.outputs.iota_ref }}" | sed 's/^v//' | sed 's/-.*$//')
```
And why exactly do we do that? So we release "release candidates" under the wrong tag as "release version"?

Dr-Electron replied:
In choco we could also do pre releases. So release whatever is stable which can be installed with `choco install iota`
and pre-releases as `choco install iota --pre`. I just wanted to release the same version as we do with homebrew.
If we keep the pre-release tag in the version choco won't let you install with `choco install iota` 🤔 Seems a little weird if we keep it like that. But not sure maybe I'm missing something. This is my first choco release

Dr-Electron replied:
Removed

muXxer commented on chocolatey/iota.nuspec:
```diff
@@ -5,23 +5,26 @@ enclosed in quotation marks, you should use an editor that supports UTF-8, not t
-		<file src="../target/release/iota.exe" target="bin/iota.exe" />
-		<file src="../LICENSE" target="bin/LICENSE.txt" />
+		<file src="../tmp/release/iota.exe" target="bin/iota.exe" />
+		<file src="../LICENSE" target="tools/LICENSE" />
```
why is the subfolder now "tools" ?

Dr-Electron replied:
That's what the choco team requested from me. The sui version actually doesn't seem to follow their rules 😅

Dr-Electron committed:
Update Chocolatey publish conditions and versioning

Dr-Electron committed:
Add comment to TMP_BUILD_DIR

thibault-martinez commented on chocolatey/iota.nuspec:
```diff
@@ -5,23 +5,26 @@ enclosed in quotation marks, you should use an editor that supports UTF-8, not t
+		<packageSourceUrl>https://github.com/iotaledger/iota/tree/develop/chocolatey</packageSourceUrl>
+		<summary>IOTA CLI to interact with the IOTA network</summary>
+		<description>Bringing the real world to Web3 with a scalable, decentralized and programmable DLT infrastructure</description>
 		<releaseNotes>https://github.com/iotaledger/iota/releases/tag/mainnet-v$version$</releaseNotes>
```
Incorrect tag

Dr-Electron replied:
Good catch fixed in f1f1694.

Dr-Electron committed:
fix: fix release notes link and simplify choco commands

Dr-Electron commented:
I fixed the wrong release notes url and also simplified the script by adding the api key to the push command and omitting the package file name as it isn't needed if there is only one in that folder

thibault-martinez committed:
Merge branch 'develop' into devx/chocolatey

thibault-martinez commented on .github/workflows/release.yml:
```diff
@@ -191,26 +192,26 @@ jobs:
-      #          choco pack --version ${{ env.iota_ref }} configuration=release
-      #          choco apikey --api-key ${{ secrets.CHOCO_API_KEY }} --source https://push.chocolatey.org/
-      #          choco push iota.${{ env.iota_ref }}.nupkg --source https://push.chocolatey.org/
+      - name: Publish Windows iota binary to Chocolatey
```
I would rather have this be its own job like homebrew because if it fails for whatever reason (e.g. network issues on their infra), it would interrupt our release process.

Dr-Electron replied:
Sure I can do that 👍

Dr-Electron committed:
Add choco push as separate job

thibault-martinez commented on .github/workflows/release.yml:
```diff
@@ -313,3 +293,52 @@ jobs:
+      - name: Set artifact name
+        id: artifact
+        shell: bash
+        run: echo "artifact_name=iota-${{ needs.generate-tag.outputs.iota_ref }}-windows-x86_64" >> $GITHUB_OUTPUT
```
`generate-tag` is not in `needs`?

Dr-Electron replied:
True. I need it 😆 
[8a05e33](https://github.com/iotaledger/iota/pull/9193/commits/8a05e3331b09534de5a0467f4ffb1e5bbd0cbb03)

Dr-Electron committed:
Require generate-tag
